File: GroceryWala_UI/src/app/Models/UserOrderModel.ts
export class UserOrderModel{
    orderId?:number;
    totalMRP: number;
    discountOnMRP: number;
    finalAmount: number;
    createdAt?: Date;
    couponCode?: string;
    extraDiscount?:number;
    userId?:string;

    constructor(totalMRP: number, discountOnMrp: number, finalAmount: number, 
        couponCode?: string, createdAt?: Date, id?:number, extraDiscount?:number){
        this.totalMRP = totalMRP;
        this.finalAmount = finalAmount;
        this.discountOnMRP = discountOnMrp;
        this.couponCode = couponCode;
        this.orderId = id;
        this.extraDiscount = extraDiscount;
        this.createdAt = createdAt;   
    }
}

File: DomainLayer/Models/Single/UserOrderModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroceryWala.DomainLayer.Models.Single
{
    public class UserOrderModel
    {
        [Key]
        public int OrderId { get; set; }

        [Required(ErrorMessage = "Required user id.")]
        public string UserId { get; set; }
        [Required(ErrorMessage = "Required total MRP.")]
        public decimal TotalMRP { get; set; }
        [Required(ErrorMessage = "Required discount on MRP.")]
        public decimal DiscountOnMRP { get; set; }
        [Required(ErrorMessage = "Required final amount.")]
        public decimal FinalAmount { get; set; }
        public string CouponCode { get; set; }
        public int ExtraDiscount { get; set; }

        public DateTime CreatedAt { get; set; }

    }
}
